thingif/src/main/java/com/kii/thingif/gateway/Gateway.java

```diff
@@ -0,0 +1,31 @@
+package com.kii.thingif.gateway;
+
+import android.os.Parcel;
+import android.support.annotation.NonNull;
+
+import com.kii.thingif.AbstractThing;
+
+public class Gateway extends AbstractThing {
+    public Gateway(@NonNull String thingID, @NonNull String vendorThingID) {
+        super(thingID, vendorThingID);
+    }
+    @Override
+    public String getAccessToken() {
```

Comment:
@satoshikumano @nfukuzaki 
I think `getAccessToken()` is not required for Gateway class. So I think `Target` interface and `ThingTarget` interface should be changed followings:

```swift
public interface Target extends Parcelable {
    @NonNull
    public TypedID getTypedID();
    @NonNull
    public String getThingID();
    @NonNull
    public String getVendorThingID();
}

public interface TargetThing extends Target {
    @Nullable
    public String getAccessToken();
}
```

I also think `AbstractTarget` class should be add and `AbstractThing` class should be modified like following:

```swift
public abstract class AbstractTarget implements Target {
    // TODO: implement me.
}

public abstract class AbstractThing extends AbstractTarget implement TargetThing {
    // TODO: implement me.
}
```

What do you think?